core/src/main/java/org/fao/geonet/kernel/oaipmh/Lib.java

```diff
@@ -152,6 +155,69 @@ public static Element toJeevesException(OaiPmhException e) {
 
         return error;
     }
+
+    static JsonNode createSearchQuery(Element params) throws JsonProcessingException {
+        final String PARAM_SET = "category";
+        final String PARAM_FROM = "extFrom";
+        final String PARAM_UNTIL = "extTo";
+        final String PARAM_METADATAPREFIX = "_schema";
+
+        String jsonQuery = "{" +
+            "    \"bool\": {" +
+            "      \"must\": [" +
+            "        {" +
+            "          \"terms\": {" +
+            "            \"isTemplate\": [\"n\"]" +
+            "          }" +
+            "        }, " +
+            "        {" +
+            "          \"term\": {" +
+            "            \"documentStandard\": {" +
+            "              \"value\": \"%s\"" +
+            "            }" +
+            "          }" +
+            "        }%s%s" +
+            "      ]" +
+            "    }" +
+            "}";
+
+        String categoryQuery = "";
+        if (params.getChild(PARAM_SET) != null) {
+            categoryQuery = String.format("        ,{" +
+                "          \"term\": {" +
+                "            \"cat\": {" +
+                "              \"value\": \"%s\"" +
+                "            }" +
+                "          }" +
+                "        }", params.getChild(PARAM_SET).getValue());
+        }
+
+        String temporalExtentQuery = "";
+        if ((params.getChild(PARAM_FROM) != null) ||
+            (params.getChild("extTo") != null)) {
+
+            String fromQuery = "";
+            if (params.getChild(PARAM_FROM) != null) {
+                fromQuery = String.format("                \"gte\": \"%s\",", params.getChild(PARAM_FROM).getValue());
+            }
+
+            String untilQuery = "";
+            if (params.getChild(PARAM_UNTIL) != null) {
+                untilQuery = String.format("                \"lte\": \"%s\",", params.getChild(PARAM_UNTIL).getValue()) ;
+            }
+
+            temporalExtentQuery = String.format("  ,{\"range\": {\"resourceTemporalDateRange\": {" +
+                "                %s" +
+                "                %s" +
+                "                \"relation\": \"intersects\"" +
+                "            }}}", fromQuery, untilQuery);
+        }
+
+        ObjectMapper objectMapper = new ObjectMapper();
+
+        String schema = (params.getChild(PARAM_METADATAPREFIX) != null) ? params.getChild(PARAM_METADATAPREFIX).getValue() : ISO19139SchemaPlugin.IDENTIFIER;
```

Comment:
This fallback seems useless to me since the request will give an error if no `metadataPrefix` parameter is given